src/language-js/needs-parens.js

```diff
@@ -334,12 +332,12 @@ function needsParens(path, options) {
       if (
         (parent.type === "TSTypeParameter" ||
           parent.type === "TypeParameter" ||
-          parent.type === "VariableDeclarator" ||
+          parent.type === "TSTypeAliasDeclaration" ||
           parent.type === "TSTypeAnnotation" ||
+          parent.type === "TSParenthesizedType" ||
           parent.type === "GenericTypeAnnotation" ||
```

Comment:
`GenericTypeAnnotation` can't be produced by estree anymore it's now always `TSTypeAnnotation`